lib/usd/ui/TreeModel.cpp

```diff
@@ -51,6 +53,49 @@ TreeItem* findTreeItem(TreeModel* treeModel, const QModelIndex& parent, std::fun
 	return nullptr;
 }
 
+void resetVariantToPrimSelection(TreeItem* variantItem)
+{
+	if (variantItem) {
+		UsdPrim prim = variantItem->prim();
+		assert(prim.IsValid() && prim.HasVariantSets());
+
+		UsdVariantSets varSets = prim.GetVariantSets();
+
+		std::vector<std::string> usdVarSetNames;
+		varSets.GetNames(&usdVarSetNames);
+
+		QStringList qtVarNames;
+		for (auto it=usdVarSetNames.crbegin(); it != usdVarSetNames.crend(); it++)
+		{
+			UsdVariantSet varSet1 = varSets.GetVariantSet(*it);
+			qtVarNames.push_back(QString::fromStdString(varSet1.GetVariantSelection()));
+		}
+
+		variantItem->setData(qtVarNames, ItemDelegate::kVariantSelectionRole);
+	}
+}
+
+void resetAllVariants(TreeModel* treeModel, const QModelIndex& parent)
+{
+	for (int r=0; r<treeModel->rowCount(parent); ++r)
+	{
+		QModelIndex variantIndex = treeModel->index(r, TreeModel::kTreeColumn_Variants, parent);
+		TreeItem* variantItem = static_cast<TreeItem*>(treeModel->itemFromIndex(variantIndex));
+		assert(variantItem);
+
+		if (variantItem->variantSelectionModified()) 
+		{
+			resetVariantToPrimSelection(variantItem);
+		}
```

Comment:
Here you assert, but then just use variantItem. So if it happened to be nullptr you would crash and thus never get to the call to resetVariantToPrimSelection where inside you have an if(variantItem) check. I would be more inclined to put the nullptr check on line 86 and move the assert into resetVariantToPrimSelection and remove the if check there. This makes resetVariantToPrimSelection a helper that is assuming valid input.

Reply:
Yeah, good point.  Will change.